### PR TITLE
Fix np1.17 isnan, isinf, isfinite ufuncs

### PR DESCRIPTION
**Pull request by sklam**
Closes https://github.com/numba/numba/issues/4319.

Numpy1.17 added specialized loops for `isnan`, `isinf` and `isfinite` for boolean and integer dtypes.  This patch adds those specialization into numba.

### CODE REVIEW

File: numba/tests/test_ufuncs.py [hunk 1]
```diff
@@ -609,17 +614,33 @@ def test_fmin_ufunc(self, flags=no_pyobj_flags):
...
+    def bool_additional_inputs(self):
+        return [
+            (np.array([True, False], dtype=np.bool_),
+             types.Array(types.bool_, 1, 'C')),
+            ]
+
     @tag('important')
     def test_isfinite_ufunc(self, flags=no_pyobj_flags):
-        self.unary_ufunc_test(np.isfinite, flags=flags)
+        self.unary_ufunc_test(
```
Comment (stuartarchibald) [inline numba/tests/test_ufuncs.py]: Should this also get a test for the `datetime` types?

**Comment by stuartarchibald** [comment]
I also tested this manually against NumPy 1.17.

File: numba/tests/test_ufuncs.py (hunk 1 above)
Comment (sklam) [reply numba/tests/test_ufuncs.py]: The datetime types are tested in `numba.tests.test_ufuncs.TestLoopTypesDatetimeNoPython`.
i.e. `python runtests.py "numba.tests.test_ufuncs.TestLoopTypesDatetimeNoPython.test_isfinite_M_?"`

Comment (stuartarchibald) [reply numba/tests/test_ufuncs.py]: Great, thanks for checking.

**Pull request merged**